### PR TITLE
Add a basic starboard

### PR DESCRIPTION
**Pull request by meooow25**
## A simple starboard cog :star: 

A starboard cog with the following functionality
- Set a particular channel as starboard.
- Reacting to a message with a :star: will post the message to a starboard channel if the reactor(?) is an admin or if the message now has >= 3 stars.
- Reacting to the same message after that has no effect on the starboard.
- Deleting the message from the starboard will allow the original message to be posted again when reacted to.

### Changes
- Added two tables, one for starboard settings and one the starboard messages.
- Added `Starboard` cog.

### Screenshot
screenshot

### CODE REVIEW

**Comment by algmyr** [comment]
If it's possible, maybe move the content to right below the image so that things look more like the original post?

File: tle/cogs/starboard.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        if payload.channel_id != starboard_channel_id:
+            return
+        cf_common.user_db.remove_starboard_message(payload.message_id)
+        self.logger.info(f'Removed message {payload.message_id} from starboard')
+
+    @staticmethod
+    def prepare_embed(message):
+        # Adapted from https://github.com/Rapptz/RoboDanny/blob/rewrite/cogs/stars.py
+        embed = discord.Embed(color=_STAR_ORANGE, timestamp=message.created_at)
+        if message.content:
```
Comment (algmyr) [inline tle/cogs/starboard.py]: As commented, would it look nice if this was below channel and link to original?

File: tle/cogs/starboard.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                    invoke_without_command=True)
+    async def starboard(self, ctx):
+        """Group for commands involving the starboard"""
+        await ctx.send_help(ctx.command)
+
+    @starboard.command(brief='Set starboard to current channel')
+    @commands.has_role('Admin')
+    async def here(self, ctx):
+        """Set the current channel as starboard"""
+        cf_common.user_db.set_starboard_settings(ctx.guild.id, ctx.channel.id)
```
Comment (algmyr) [inline tle/cogs/starboard.py]: Can this add multiple starboards? If so, is this intended or will it lead to weird behavior? If not, maybe add a check and show an error.

File: tle/cogs/starboard.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            embed.add_field(name='Content', value=message.content, inline=False)
+
+        if message.embeds:
+            data = message.embeds[0]
+            if data.type == 'image':
+                embed.set_image(url=data.url)
+
+        if message.attachments:
+            file = message.attachments[0]
+            if file.url.lower().endswith(('png', 'jpeg', 'jpg', 'gif', 'webp')):
```
Comment (algmyr) [inline tle/cogs/starboard.py]: could add webm for completeness

Comment (meooow25) [reply tle/cogs/starboard.py]: Sadly, Discord can embed videos but we cannot :(

File: tle/cogs/starboard.py (hunk 1 above)
Comment (meooow25) [reply tle/cogs/starboard.py]: Agreed, and done.

File: tle/cogs/starboard.py (hunk 2 above)
Comment (meooow25) [reply tle/cogs/starboard.py]: Discussed and resolved.

**Pull request merged**